tests/budget_accounting_test.py

```diff
@@ -48,5 +53,147 @@ def test_compute_budgets(self):
+                    ComputeBudgetMechanisms(5, 50, NoiseKind.GAUSSIAN, 1, 1)
+                ],
+                expected_pipeline_noise_std=50),
+            ComputeBudgetTestCase(
+                name="multiple_weights",
+                expected_epsilon=1.924852037917208,
+                delta=1e-5,
+                mechanisms=[
+                    ComputeBudgetMechanisms(4, 10, NoiseKind.LAPLACE, 2, 1),
+                    ComputeBudgetMechanisms(4, 5, NoiseKind.GAUSSIAN, 4, 1)
+                ],
+                expected_pipeline_noise_std=20),
+            ComputeBudgetTestCase(
+                name="multiple_sensitivities",
+                expected_epsilon=0.2764312848667339,
+                delta=1e-5,
+                mechanisms=[
+                    ComputeBudgetMechanisms(6, 40, NoiseKind.LAPLACE, 1, 2),
+                    ComputeBudgetMechanisms(2, 80, NoiseKind.GAUSSIAN, 1, 4)
+                ],
+                expected_pipeline_noise_std=20),
+            ComputeBudgetTestCase(
+                name="multiple_weights_and_sensitivities",
+                expected_epsilon=0.780797891312483,
+                delta=1e-5,
+                mechanisms=[
+                    ComputeBudgetMechanisms(4, 10, NoiseKind.LAPLACE, 4, 2),
+                    ComputeBudgetMechanisms(6, 40, NoiseKind.GAUSSIAN, 2, 4)
+                ],
+                expected_pipeline_noise_std=20),
+            ComputeBudgetTestCase(
+                name="multiple_weights_and_sensitivities_variants",
+                expected_epsilon=0.9165937807680077,
+                delta=1e-6,
+                mechanisms=[
+                    ComputeBudgetMechanisms(4, 20, NoiseKind.LAPLACE, 4, 2),
+                    ComputeBudgetMechanisms(6, 80, NoiseKind.GAUSSIAN, 2, 4),
+                    ComputeBudgetMechanisms(1, 80, NoiseKind.GAUSSIAN, 3, 6),
+                    ComputeBudgetMechanisms(5, 15, NoiseKind.LAPLACE, 8, 3),
+                ],
+                expected_pipeline_noise_std=40)
+        ]
+
+        for case in testcases:
+            accountant = PLDBudgetAccountant(case.expected_epsilon, case.delta,
+                                             1e-2)
+            actual_mechanisms = []
+            for mechanism in case.mechanisms:
+                for _ in range(0, mechanism.count):
+                    actual_mechanisms.append(
+                        (mechanism.expected_noise_std,
+                         accountant.request_budget(
+                             mechanism.noise_kind,
+                             weight=mechanism.weight,
+                             sensitivity=mechanism.sensitivity)))
+            self.assertEqual(
+                len(actual_mechanisms), len(accountant._mechanisms),
+                f"failed test {case.name} expected len {len(actual_mechanisms)} "
+                f"got len {len(accountant._mechanisms)}")
+            compare_pld = accountant._compose_distributions(
```

Comment:
I like that approach to test _compose_distributions and compute_budgets on the same data